### PR TITLE
apps/testcase/systemio : uart, gpio ITC refactor

### PR DESCRIPTION
**Pull request by arvinmittal**
Refactor of ITCs of SystemIO : uart and gpio
TC Naming rule is <TC_Category>_<Module>_<SubModule>_<FunctionName>_<Positive/Negative>_<AdditionalInformation>
Current TC names do not have above naming convention
Remove not required header files and correct file name comment
Restructure TCs and declare variable in seperate line.
Current TC do not have this conventtion

Signed-off-by: Arvin Mittal <arvin.mittal@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c [hunk 1]
```diff
@@ -20,295 +20,325 @@
...
-#include <tinyara/config.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <errno.h>
-#include <tinyara/time.h>
-#include <sys/time.h>
 #include "itc_internal.h"
 #include <iotbus_gpio.h>
-#include <iotbus_error.h>
 #ifdef CONFIG_ARCH_CHIP_STM32
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c]: To use CONFIG_XX, you should keep including <tinyara/config.h>

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c [hunk 2]
```diff
@@ -20,295 +20,325 @@
...
 #include "../../os/arch/arm/src/mikroequail/src/mikroequail_gpio.h"
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c]: We don't have this file. Let's remove.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c [hunk 3]
```diff
@@ -16,20 +16,13 @@
...
 
-/// @file iotbus_uart.h
+/// @file itc_uart.c
 
 /// @brief Test Case Iotbus APIs for UART
 
 /****************************************************************************
  * Included Files
  ****************************************************************************/
-#include <tinyara/config.h>
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: same as previous

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c (hunk 1 above)
Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c]: Okay, I included <tinyara/config.h>

File: apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c (hunk 3 above)
Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_uart.c]: Okay, I included <tinyara/config.h>

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c (hunk 2 above)
Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c]: Okay, removed

**Comment by arvinmittal** [comment]
@sunghan-chang 
Fixed your review comments. Please verify.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**